app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -741,31 +754,32 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
         startActivityForResult(speechIntent, VoiceSearchActivity.SPEECH_REQUEST_CODE)
     }
 
-    private fun addQrButton(searchFragmentState: SearchFragmentState) {
-        if (qrButtonAlreadyAdded) {
-            return
-        }
-
-        val searchEngine = searchFragmentState.searchEngineSource.searchEngine
-
-        val isVisible =
-            searchEngine?.id?.contains("google") == true &&
-                isSpeechAvailable() &&
-                requireContext().settings().shouldShowVoiceSearch
+    private fun updateQrButton(searchFragmentState: SearchFragmentState) {
+        when (searchFragmentState.searchEngineSource.searchEngine == searchFragmentState.defaultEngine) {
+            true -> {
+                if (qrButtonAction == null) {
+                    val qrAction = IncreasedTapAreaActionDecorator(
```

Comment:
You can do this for `voiceSearchButtonAction` as well, but I would just assign this to `qrButtonAction` and remove `qrAction` and save a line of code for the assigning.

Reply:
I tried that originally but got a warning of "Smart cast to 'IncreasedTapAreaActionDecorator' is impossible, because 'qrButtonAction' is a mutable property that could have been changed by this time".  Had to set it to a val so it works.